dotCMS/src/integration-test/java/com/dotcms/security/apps/AppsAPIImplTest.java

```diff
@@ -418,6 +434,160 @@ public void Test_Secret_Destroy_Method() {
+                        "icon", false,
+                        emptyParams),
+                //The following test paramDefinition.
+                //Null type  is not allowed.
+                new AppTestCase("any-key", "any-name", "desc", "icon", false,
+                       ImmutableSortedMap.of("p1", newParam("", true, null, "", "", true)
+                )),
+                //Hidden bool param is not allowed.
+                new AppTestCase("any-key", "any-name", "desc", "icon", false,
+                       ImmutableSortedMap.of("p1", newParam("", true, Type.BOOL, "label", "hint", true)
+                )),
+                //emptyLabel.
+                new AppTestCase("any-key", "any-name", "desc", "icon", false,
+                       ImmutableSortedMap.of("p1", newParam("v1", true, Type.STRING, "", "", true)
+                )),
+                //emptyHint.
+                new AppTestCase("any-key", "any-name", "desc", "icon", false,
+                       ImmutableSortedMap.of("p1", newParam("v1", true, Type.STRING, "label", "", true)
+                )),
+                //Required param with null default.
+                new AppTestCase("any-key", "any-name", "desc", "icon", false,
+                       ImmutableSortedMap.of("p1", newParam("null", false, Type.STRING, "label", "hint", true)
+                )),
+                //non parsable to bool string.
+                new AppTestCase("any-key", "any-name", "desc", "icon", false,
+                        ImmutableSortedMap.of("p1", newParam("lol", false, Type.BOOL, "label", "hint", true)
+                )),
+                //Null hidden to emulate missing hidden field.
+                new AppTestCase("any-key", "any-name", "desc", "icon", false,
+                        ImmutableSortedMap.of(
+                                "p1", newParam("false", null, Type.BOOL, "label", "hint", true)
+                        )),
+                //Null required to emulate missing hidden field.
+                new AppTestCase("any-key", "any-name", "desc", "icon", false,
+                        ImmutableSortedMap.of(
+                                "p1", newParam("false", false, Type.BOOL, "label", "hint", null)
+                        ))
+        };
+    }
+
+    /**
+     * Test AppsAPI#createAppDescriptor input validation.
+     * Given scenario: Passing Params that normally wouldn't break the validation rules.
+     * Expected Result: No DotDataValidationException is raised.
+     */
+    @Test
+    @UseDataProvider("getValidExceptionFreeTestCases")
+    public void Test_App_Descriptor_Validation_Exception_Free(final AppTestCase testCase)
+            throws IOException, DotDataException, DotSecurityException {
+        assertNotNull(evaluateAppTestCase(testCase));
+    }
+
+
+    @DataProvider
+    public static Object[] getValidExceptionFreeTestCases() throws Exception {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5358201)